### PR TITLE
Create new project folder for a new project

### PR DESCRIPTION
**Pull request by nakhbari**
Fixes bug where we check the environment config for all projects even when the folder does not exist yet.

### CODE REVIEW

File: app/services/config_data_sources/json_project_data_source.rb [hunk 1]
```diff
@@ -185,6 +185,9 @@ def create_project!(
         artifact_provider: artifact_provider,
         job_triggers: job_triggers
       )
+      
+      FileUtils.mkdir_p(File.join(json_folder_path, "projects", new_project.id))
```
[the remark this answers is not in the document]
Comment (nakhbari) [reply app/services/config_data_sources/json_project_data_source.rb]: I'm not sure how to do that since a data source could be for any folder. ProjectDataSource specifically needs the project folder. Also would have to pipe the project ID to the DataSource somehow. 

Or should i make a required method for the "resource path"?

[the remark this answers is not in the document]
Comment (taquitos) [reply app/services/config_data_sources/json_project_data_source.rb]: Yup, I have a couple of tasks I need to finish up before my battery dies 💀 so I think I'll have to get to it tomorrow

[the remark this answers is not in the document]
Comment (taquitos) [reply app/services/config_data_sources/json_project_data_source.rb]: I think you're right, I did some digging, and there are a few things at play (which you mentioned) the `json_folder_path` needs a `projects` folder. I was thinking this could be automatically created but it turns out where we instantiate a `JSONProjectDataSource` we only pass in `ci_config_git_repo_path` so we don't yet specify that we need a `projects` subfolder. Later on we do, and so we can't actually add that to the `JSONDataSource` without adding more complexity that doesn't benefit us.

TL;DR - 🎷:goat:
